demovieworderhooks/demovieworderhooks.php

```diff
@@ -0,0 +1,222 @@
+        return 'displayAdminOrderMain';
+    }
+
+    /**
+     * Displays customer's review about the order.
+     */
+    public function hookDisplayAdminOrderSide(array $params)
+    {
+        /** @var OrderReviewRepository $orderReviewRepository */
+        $orderReviewRepository = $this->get('prestashop.module.demovieworderhooks.repository.order_review_repository');
+
+        /** @var OrderReviewPresenter $orderReviewPresenter */
+        $orderReviewPresenter = $this->get('prestashop.module.demovieworderhooks.presenter.order_review_presenter');
+
+        $orderReview = $orderReviewRepository->findOneBy(['orderId' => $params['id_order']]);
+
+        if (!$orderReview) {
+            return '';
+        }
+
+        return $this->render($this->getModuleTemplatePath() . 'customer_satisfaction.html.twig', [
+            'orderReview' => $orderReviewPresenter->present($orderReview),
+        ]);
+    }
+
+    /**
+     * Displays other orders from the same customer in a block.
+     */
+    public function hookDisplayAdminOrder(array $params)
+    {
+        /** @var OrderRepository $orderRepository */
+        $orderRepository = $this->get('prestashop.module.demovieworderhooks.repository.order_repository');
+
+        /** @var OrdersPresenter $ordersPresenter */
+        $ordersPresenter = $this->get('prestashop.module.demovieworderhooks.presenter.orders_presenter');
+
+        $order = new Order($params['id_order']);
+
+        return $this->render($this->getModuleTemplatePath() . 'customer_orders.html.twig', [
+            'currentOrderId' => (int) $params['id_order'],
+            'orders' => $ordersPresenter->present(
+            // Get all customer orders except currently viewed order
+                $orderRepository->getCustomerOrders((int) $order->id_customer, [$order->id]),
+                (int) $this->context->language->id
+            ),
+        ]);
+    }
+
+    /**
+     * Displays previous/next order buttons.
+     */
+    public function hookDisplayAdminOrderTop(array $params)
+    {
+        /** @var OrderRepository $orderRepository */
+        $orderRepository = $this->get('prestashop.module.demovieworderhooks.repository.order_repository');
+
+        /** @var OrderLinkPresenter $orderLinkPresenter */
+        $orderLinkPresenter = $this->get('prestashop.module.demovieworderhooks.presenter.order_link_presenter');
+
+        $nextOrderId = $orderRepository->getNextOrderId((int) $params['id_order']);
+        $previousOrderId = $orderRepository->getPreviousOrderId((int) $params['id_order']);
+
+        return $this->render($this->getModuleTemplatePath() . 'order_navigation.html.twig', [
+            'previousOrder' => $orderLinkPresenter->present($previousOrderId),
+            'nextOrder' => $orderLinkPresenter->present($nextOrderId),
+        ]);
+    }
+
+    public function hookActionGetAdminOrderButtons(array $params)
+    {
+        // export order
+        return 'actionGetAdminOrderButtons';
```

Comment:
This one is tricker, we should return an ActionsBarButtonsCollection to be able to use it properly

See https://github.com/matks/orderhooks/blob/master/orderhooks.php#L49

Reply:
This was is only used in legacy controller so will remove it